### PR TITLE
avx2 add, broadcast, reg copy instructions

### PR DESCRIPTION
**Pull request by SamirDroubi**

### CODE REVIEW

**Comment by yamaguchi1024** [comment]
@alexreinking can you take a quick look? This looks good to me

File: tests/test_x86.py [hunk 1]
```diff
@@ -598,3 +598,192 @@ def run_and_check(x, result):
...
+    val_copy = val.copy()
+
+    getattr(fn, "mm256_broadcast_ss_scalar_wrapper")(None, out, val)
+    getattr(fn, "mm256_broadcast_ss_scalar_ref")(None, out_copy, val_copy)
+
+    np.testing.assert_almost_equal(out, out_copy)
+    np.testing.assert_almost_equal(val, val_copy)
+
+
+def test_mm256_add_ps(compiler):
```
Comment (alexreinking) [inline tests/test_x86.py:651]: These tests need to be marked with `@pytest.mark.isa("AVX2")`

**Review by alexreinking — CHANGES_REQUESTED** [review]

File: tests/test_x86.py (hunk 1 above)
Comment (alexreinking) [reply tests/test_x86.py:651]: See here for an example:

https://github.com/exo-lang/exo/blob/c56d978350bc5a524d2bd932fd9ae4e313a0f1ce/tests/test_x86.py#L23-L40

Comment (SamirDroubi) [reply tests/test_x86.py:653]: Done

**Review by alexreinking — APPROVED** [review]

File: src/exo/platforms/x86.py [hunk 2]
```diff
@@ -99,6 +99,15 @@ def mm256_add_ps(out: [f32][8] @ AVX2, x: [f32][8] @ AVX2, y: [f32][8] @ AVX2):
         out[i] = x[i] + y[i]
 
 
+@instr("{dst_data} = _mm256_add_ps ({src_data}, {dst_data});")
```
Comment (alexreinking) [inline src/exo/platforms/x86.py]: We should keep the generated code consistently formatted.
```suggestion
@instr("{dst_data} = _mm256_add_ps({src_data}, {dst_data});")
```

**Review by alexreinking — CHANGES_REQUESTED** [review]

File: src/exo/platforms/x86.py [hunk 3]
```diff
@@ -281,3 +308,21 @@ def avx2_assoc_reduce_add_ps(x: [f32][8] @ AVX2, result: f32):
...
+@instr("{dst_data} = _mm256_mul_ps({src_data}, _mm256_set1_ps(-1.0f));")
+def avx2_sign_ps(dst: [f32][8] @ AVX2, src: [f32][8] @ AVX2):
+    assert stride(dst, 0) == 1
+    assert stride(src, 0) == 1
+
+    for i in seq(0, 8):
+        dst[i] = -src[i]
+
+
+@instr("{dst_data} = _mm256_add_ps ({src_data}, {dst_data});")
```
Comment (alexreinking) [inline src/exo/platforms/x86.py:322]:
```suggestion
@instr("{dst_data} = _mm256_add_ps({src_data}, {dst_data});")
```
Same here

**Review by alexreinking — APPROVED** [review]

File: src/exo/platforms/x86.py (hunk 2 above)
Comment (SamirDroubi) [reply src/exo/platforms/x86.py]: Done

**Pull request merged**